File: db/java/app/src/main/java/com/example/sales/domain/model/Stock.java
package com.example.sales.domain.model;

import java.time.LocalDateTime;

/**
 * 在庫データのEntityクラス
 */
public class Stock {
    private String warehouseCode;
    private String productCode;
    private String lotNo;
    private String stockType;
    private String qualityType;
    private Integer actualQuantity;
    private Integer validQuantity;
    private LocalDateTime lastDeliveryDate;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    // デフォルトコンストラクタ
    public Stock() {}

    // Getter/Setter（全フィールド）
    public String getWarehouseCode() { return warehouseCode; }
    public void setWarehouseCode(String warehouseCode) {
        this.warehouseCode = warehouseCode;
    }

    public String getProductCode() { return productCode; }
    public void setProductCode(String productCode) { this.productCode = productCode; }

    public String getLotNo() { return lotNo; }
    public void setLotNo(String lotNo) { this.lotNo = lotNo; }

    public String getStockType() { return stockType; }
    public void setStockType(String stockType) { this.stockType = stockType; }

    public String getQualityType() { return qualityType; }
    public void setQualityType(String qualityType) { this.qualityType = qualityType; }

    public Integer getActualQuantity() { return actualQuantity; }
    public void setActualQuantity(Integer actualQuantity) {
        this.actualQuantity = actualQuantity;
    }

    public Integer getValidQuantity() { return validQuantity; }
    public void setValidQuantity(Integer validQuantity) {
        this.validQuantity = validQuantity;
    }

    public LocalDateTime getLastDeliveryDate() { return lastDeliveryDate; }
    public void setLastDeliveryDate(LocalDateTime lastDeliveryDate) {
        this.lastDeliveryDate = lastDeliveryDate;
    }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public String getCreatedBy() { return createdBy; }
    public void setCreatedBy(String createdBy) { this.createdBy = createdBy; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    public String getUpdatedBy() { return updatedBy; }
    public void setUpdatedBy(String updatedBy) { this.updatedBy = updatedBy; }
}

File: db/go/sales_management/internal/model/stock.go
package model

import (
	"database/sql"
	"time"
)

// Stock は在庫データのドメインモデルです
// 倉庫コード、商品コード、ロット番号、在庫区分、品質区分の5つのフィールドで主キーを構成
type Stock struct {
	WarehouseCode    string       `db:"倉庫コード"`
	ProductCode      string       `db:"商品コード"`
	LotNumber        string       `db:"ロット番号"`
	StockCategory    string       `db:"在庫区分"`  // 1:通常在庫、2:預かり在庫、3:委託在庫
	QualityCategory  string       `db:"品質区分"`  // A:良品、B:不良品、C:検査中
	ActualStock      int          `db:"実在庫数"`  // 物理的に存在する在庫数
	AvailableStock   int          `db:"有効在庫数"` // 販売可能な在庫数
	LastShippingDate sql.NullTime `db:"最終出荷日"`
	CreatedAt        time.Time    `db:"作成日時"`
	CreatedBy        string       `db:"作成者名"`
	UpdatedAt        time.Time    `db:"更新日時"`
	UpdatedBy        string       `db:"更新者名"`
}
